dspace-spring-rest/src/test/java/org/dspace/app/rest/RelationshipTypeRestControllerIT.java

```diff
@@ -0,0 +1,268 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.io.File;
+import java.util.Iterator;
+import java.util.List;
+
+import org.dspace.app.rest.builder.CollectionBuilder;
+import org.dspace.app.rest.builder.CommunityBuilder;
+import org.dspace.app.rest.builder.ItemBuilder;
+import org.dspace.app.rest.builder.RelationshipBuilder;
+import org.dspace.app.rest.matcher.EntityTypeMatcher;
+import org.dspace.app.rest.matcher.PageMatcher;
+import org.dspace.app.rest.matcher.RelationshipMatcher;
+import org.dspace.app.rest.matcher.RelationshipTypeMatcher;
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.EntityType;
+import org.dspace.content.Item;
+import org.dspace.content.Relationship;
+import org.dspace.content.RelationshipType;
+import org.dspace.content.service.EntityTypeService;
+import org.dspace.content.service.RelationshipService;
+import org.dspace.content.service.RelationshipTypeService;
+import org.dspace.services.ConfigurationService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class RelationshipTypeRestControllerIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    private RelationshipTypeService relationshipTypeService;
+
+    @Autowired
+    private EntityTypeService entityTypeService;
+
+    @Autowired
+    private RelationshipService relationshipService;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Before
+    public void setup() throws Exception {
+
+        //Set up the database for the next test
+        String pathToFile = configurationService.getProperty("dspace.dir") +
+            File.separator + "config" + File.separator + "entities" + File.separator + "relationship-types.xml";
+        runDSpaceScript("initialize-entities", "-f", pathToFile);
+
+    }
+
+    @After
+    public void destroy() throws Exception {
+        //Clean up the database for the next test
```

Comment:
The act of creating/recreating all relationship types *for every test* is going to cause this Integration Test class to run very slowly.  Do we really need to initialize all relationship-types over and over again?   Shouldn't they just be initialized once (on startup)?

Also, similar to my past comments on the CsvImportIT, this IT doesn't need to be cleaning up every object manually (which is prone to mistakes if you forget to cleanup something).  Instead, it should be using the established `AbstractBuilder` model that all other REST API ITs use.  See this ticket https://jira.duraspace.org/browse/DS-4121

Reply:
This will be dealt with in the separate ticket as discussed in the last meeting